### PR TITLE
fixed bug where mutliple pulse inits will crash panon

### PR DESCRIPTION
**Pull request by yuannan**
also minor optimisation to client.py

### CODE REVIEW

**Comment by yuannan** [comment]
Don't mind the commit history, it's a bit messy since I forgot to pull first. Just check the file changed tab. Just a few lines changed. :)

File: panon/backend/source.py [hunk 1]
```diff
@@ -119,7 +120,6 @@ def read(self, fps=None):
         return data
 
     def update_smart_device(self, ):
-        p = sc.pulseaudio._PulseAudio()
         name = p.server_info['default sink id']
```
Comment (rbn42) [inline panon/backend/source.py:123]: Are you sure server_info will be updated if line 122 is removed?

File: panon/backend/client.py [hunk 2]
```diff
@@ -72,17 +72,15 @@ async def mainloop():
...
-                        spectrum_source.update_smart_device()
+            if smart_mode and spectrum_source.smart_device_id == '':
+                spectrum_source.update_smart_device()
 
             if not use_glDFT and spectrum_data is None:
                 # Set fps to 2 to lower CPU usage, when audio is unavailable.
                 latest_wave_data = spectrum_source.read(fps=2)
-                if type(spectrum_source) is source.SoundCardSource:
-                    spectrum_source.update_smart_device()
+                spectrum_source.smart_device_id = ""
```
Comment (rbn42) [inline panon/backend/client.py:83]: When muted, this line will cause a Microphone object to be destroyed and created again and again (in update_smart_device()) in every cycle.

Comment (yuannan) [reply panon/backend/client.py:83]: Ah yes, I've noticed this too. It's more than just a visual glitch as well.

For example pauvcontrol will jitter out of control, some other app also don't like it and could cause crashes.

**Pull request merged**